src/array/reducer/swap-control.spec.ts

```diff
@@ -0,0 +1,70 @@
+import { SwapArrayControlAction } from '../../actions';
+import { createFormArrayState } from '../../state';
+import { moveControlReducer } from './move-control';
+import { swapControlReducer } from './swap-control';
+import { FORM_CONTROL_0_ID, FORM_CONTROL_ID, INITIAL_STATE_NESTED_GROUP } from './test-util';
+
+describe(`form array ${swapControlReducer.name}`, () => {
+  const testArrayValue = [0, 1, 2, 3, 4, 5];
+  const testArrayState = createFormArrayState(FORM_CONTROL_ID, testArrayValue);
+
+  it('should swap controls forwards', () => {
+    const action = new SwapArrayControlAction(FORM_CONTROL_ID, 0, 2);
+    const resultState = swapControlReducer(testArrayState, action);
+    expect(resultState.value).toEqual([2, 1, 0, 3, 4, 5]);
+  });
+
+  it('should swap controls backwards', () => {
+    const action = new SwapArrayControlAction(FORM_CONTROL_ID, 5, 1);
+    const resultState = swapControlReducer(testArrayState, action);
+    expect(resultState.value).toEqual([0, 5, 2, 3, 4, 1]);
+    expect(resultState.isDirty).toEqual(true);
+
+  });
+
+  it('should throw on out of bound or negative indices', () => {
```

Comment:
The test name mentions negative indices, but the test doesn't contain code to check for negative indices. If it did it would reveal a bug since the implementation does not check for negative indices.